InnerEye/ML/dataset/scalar_dataset.py

```diff
@@ -31,57 +31,87 @@
+    If num_classes is greater than 1 (multilabel datasets):
+        The function maps a pipe-separated set of classes to a tensor with ones at the indices
+        of the positive classes and 0 elsewhere (for example if we have a task with 6 label classes,
+        map "1|3|4" to [0, 1, 0, 1, 1, 0]).
+        If the entry in the CSV file was missing (no string given at all) or an empty string,
+        this function returns an all-zero tensor (none of the label classes were positive for this sample).
+
+    For regression datasets:
     The function casts a string label to float. Raises an exception if the conversion is
     not possible.
-    If the entry in the CSV file was missing (no string given at all), it returns math.nan.
+    If the entry in the CSV file was missing (no string given at all) or an empty string, it returns math.nan.
+
     :param label_string: The value of the label as read from CSV via a DataFrame.
     :param sample_id: The sample ID where this label was read from. This is only used for creating error messages.
-    :return:
+    :param num_classes: Number of classes. This should be equal the size of the model output.
+    For binary classification tasks, num_classes should be one. For multilabel classification tasks, num_classes should
+    correspond to the number of label classes in the problem.
+    :param is_classification_dataset: If the model is a classification model
+    :return: A list of floats with the same size as num_classes
     """
+
+    if num_classes < 1:
+        raise ValueError(f"Subject {sample_id}: Invalid number of classes: '{num_classes}'")
+
     if isinstance(label_string, float):
         if math.isnan(label_string):
-            # When loading a dataframe with dtype=str, missing values can be encoded as NaN, and get into here.
-            return label_string
+            if num_classes == 1:
+                # Pandas special case: When loading a dataframe with dtype=str, missing values can be encoded as NaN, and get into here.
+                return [label_string]
+            else:
+                return [0] * num_classes
         else:
-            raise ValueError(f"Subject {sample_id}: Unexpected float input {label_string} - did you read the "
+            raise ValueError(f"Subject {sample_id}: Unexpected float input {label_string} - did you read the "	
                              f"dataframe column as a string?")
-    if label_string:
+
+    if not label_string:
+        if not is_classification_dataset or num_classes == 1:
+            return [math.nan]
+        else:
+            return [0] * num_classes
+
+    if is_classification_dataset:
+        if num_classes == 1:
+            label_lower = label_string.lower()
+            if label_lower in ["true", "yes"]:
+                return [1.0]
+            elif label_lower in ["false", "no"]:
+                return [0.0]
+            elif label_string in ["0", "1"]:
+                return [float(label_string)]
```

Comment:
Should we allow other values that can be cast to `float`, like `NaN`? This can be useful for when we have missing labels, for example.